junit-platform-commons/src/main/java/org/junit/platform/commons/util/ResourceFileVisitor.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2015-2024 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.commons.util;
+
+import static java.nio.file.FileVisitResult.CONTINUE;
+
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.function.Consumer;
+
+import org.junit.platform.commons.logging.Logger;
+import org.junit.platform.commons.logging.LoggerFactory;
+
+/**
+ * @since 1.11
+ */
+class ResourceFileVisitor extends SimpleFileVisitor<Path> {
+
+	private static final Logger logger = LoggerFactory.getLogger(ResourceFileVisitor.class);
+
+	static final String CLASS_FILE_SUFFIX = ".class";
+
+	private final Consumer<Path> classFileConsumer;
+
+	ResourceFileVisitor(Consumer<Path> classFileConsumer) {
+		this.classFileConsumer = classFileConsumer;
+	}
+
+	@Override
+	public FileVisitResult visitFile(Path file, BasicFileAttributes attributes) {
+		if (isNotClassFile(file)) {
```

Comment:
I'm working with the assumption that when searching for test resources, we are not interested in classes and can exclude from discovery as soon as possible. I'm not sure if this assumption is correct.